python/lsst/ts/m2gui/controltab/tab_ilc_status.py

```diff
@@ -0,0 +1,285 @@
+from qasync import asyncSlot
+
+from ..utils import (
+    create_grid_layout_buttons,
+    create_group_box,
+    create_label,
+    set_button,
+)
+from . import TabDefault
+
+
+class TabIlcStatus(TabDefault):
+    """Table of the inner-loop controller (ILC) status.
+
+    Parameters
+    ----------
+    title : `str`
+        Table's title.
+    model : `Model`
+        Model class.
+
+    Attributes
+    ----------
+    model : `Model`
+        Model class.
+    """
+
+    def __init__(self, title, model):
+        super().__init__(title, model)
+
+        # Indicators of the ILCs
+        self._indicators_ilc = self._create_indicators_ilc(NUM_INNER_LOOP_CONTROLLER)
+
+        # Internal layout
+        self.widget().setLayout(self._create_layout())
+
+        # Resize the dock to have a similar size of layout
+        self.resize(self.widget().sizeHint())
+
+        # Set the callback of signal
+        self._set_signal_ilc_status(self.model.signal_ilc_status)
+
+    def _create_indicators_ilc(self, number):
+        """Creates indicators for the inner-loop controller (ILC).
+
+        Parameters
+        ----------
+        number : `int`
+            Total number of ILC.
+
+        Returns
+        -------
+        indicators : `list`
+            Indicators of ILC.
+        """
+
+        indicators = list()
+
+        # ModBUS ID begins from 1 instead of 0
+        for specific_id in range(1, number + 1):
+            indicator = set_button(
+                str(specific_id), None, is_indicator=True, is_adjust_size=True
+            )
+
+            self._update_indicator_color(indicator, InnerLoopControlMode.Unknown)
+
+            indicators.append(indicator)
+
+        return indicators
+
+    def _update_indicator_color(self, indicator, mode):
+        """Update the color of indicator.
+
+        Parameters
+        ----------
+        indicator : `PySide2.QtWidgets.QPushButton`
+            Indicator.
+        mode : enum `lsst.ts.m2com.InnerLoopControlMode`
+            Mode of inner-loop controller (ILC).
+        """
+
+        palette = indicator.palette()
+        palette.setColor(QPalette.Button, self._get_indicator_color(mode))
+
+        indicator.setPalette(palette)
+
+    def _get_indicator_color(self, mode):
+        """Get the indicator color based on the mode.
+
+        Parameters
+        ----------
+        mode : enum `lsst.ts.m2com.InnerLoopControlMode`
+            Mode of inner-loop controller (ILC).
+
+        Returns
+        -------
+        `PySide2.QtCore.Qt.GlobalColor`
+            Indicator color.
+        """
+
+        if mode == InnerLoopControlMode.Standby:
+            return Qt.cyan
+
+        elif mode == InnerLoopControlMode.Disabled:
+            return Qt.blue
+
+        elif mode == InnerLoopControlMode.Enabled:
+            return Qt.green
+
+        elif mode == InnerLoopControlMode.Fault:
+            return Qt.red
+
+        else:
+            return Qt.gray
```

Comment:
I tried this and realized that I could not decode the color's name in `TabIlcStatus._create_label_mode()` by `self._get_indicator_color(state).name.decode()` because the returned data type is `PySide2.QtGui.QColor.fromRgbF(0.937255, 0.937255, 0.937255, 1.000000)` instead of `PySide2.QtCore.Qt.GlobalColor.gray`. Therefore, I would seep the original implementation. Thanks!